### DISCUSSION TITLE
How do I make MAME remember to use UNIBIOS for Neo Geo games?

### DISCUSSION THREAD

**Discussion by TsukiZero** [mamedev/discussions#125] [Q&A]
I tried setting UNIBIOS for Metal Slug 2 and it does load it... ONCE. As soon as I return to the MAME menu, it only loads it with the default EUROPEAN Neo Geo Arcade Bios from then on. It applies to any game I set to use UNIBIOS.

**Comment by einstein95** [comment]
Start the game using `-bios unibios40 -writeconfig`, then you won't need to specify the bios from then on as the setting will be written to the game's .ini file.

**Comment by TsukiZero** [comment]
It fixes nothing: after that, if I use the GUI to start the game, the first start works as intended, the second and later IGNORES the BIOS setting UNTIL the GUI is closed.
I also did the `neogeo.ini` with `bios unibios40` thing, but really is a GUI issue.

**Comment by sairuk** [comment]
create a `mslug2.ini` (or `neogeo.ini` if you prefer)
1. if you don't already have one create a `mame.ini` with `mame -cc` 
2. configure the rompaths in the `mame.ini` file
3. copy the `mame.ini` to your `<inipath>/mslug2.ini`
4. edit `mslug2.ini` and set the value for bios to `unibios40`

disable the bios selection menu
1. edit `ui.ini` and turn on `skip_biosmenu`

disabling the bios menu is a global setting

**Reply by TsukiZero** [reply]
> create a `mslug2.ini`
> 
>     1. if you don't already have one create a `mame.ini` with `mame -cc`
> 
>     2. configure the rompaths in the `mame.ini` file
> 
>     3. copy the `mame.ini` to your `<inipath>/mslug2.ini`
> 
>     4. edit `mslug2.ini` and set the value for bios to `unibios40`
> 
> 
> disable the bios selection menu
> 
>     1. edit `ui.ini` and turn on `skip_biosmenu`
> 
> 
> disabling the bios menu is a global setting

Same issue: The GUI only uses the configured BIOS once, then reverts on load to the default, unless I close the GUI and open again whenever I want to change the game.

**Reply by sairuk** [reply]
then you've probably run into this https://github.com/mamedev/mame/issues/11313

**Reply by TsukiZero** [reply]
> then you've probably run into this [mamedev/mame#11313](https://github.com/mamedev/mame/issues/11313)

Certainly that, yes, and for some reason it hasn't been fixed yet, huh...
That's frustrating.

**Comment by sairuk** [comment]
one other thing to mention is the system ini file (e.g. `neogeo.ini` or `mslug2.ini`) only needs to contain the changed options.  

It does not need to be a complete replica of the `mame.ini`, each ini+option is an override (not an append) to the previous so your main config can hold general paths and will be used if you omit that setting from the `neogeo.ini` ... as such you can override as required per system

for example to use the unibios40 my `neogeo.ini` only contains the `bios` option
```
$ cat neogeo.ini 
bios                      unibios40
$
```

https://docs.mamedev.org/advanced/multiconfig.html

although it is worth mentioning (as it has caught out a number of people) that "Save System Settings" from the UI currently saves an entirely new configuration ini for the system with all defaults plus your changes in the UI System menu but does not carry over the rompath config from `mame.ini` unfortunately.

i am mentioning the above for folks trying to understand the behaviour, it does not work around https://github.com/mamedev/mame/issues/11313

**Comment by JimCarlTay** [comment]
MAME User Interface:

1. Open MAME.
2. Set the search filter to "Available".
3. Search "mslug2" or your desired Neo Geo game by its ROM set name.
4. Press Esc to clear the search query.
5. Double-click "System Settings" and set the system BIOS to UniBIOS ver. 4.0.
6. Double-click "Save System Settings" and double-click "Return to Previous Menu".
7. Double-click "General Settings" and double-click "Miscellaneous Options".
8. Ensure "Skip BIOS selection menu" is enabled. If not, set it to On to enable it.
9. Double-click "Return to Previous Menu".
10. Double-click "Save Settings" and double-click "Return to Previous Menu".
11. Load the game and ensure that the correct BIOS is loaded (in this case, UniBIOS ver. 4.0).

Remember that you will need to do it for each individual Neo Geo game (as arcade ROM sets).
